Sources/ZIPFoundation/Archive+Writing.swift

```diff
@@ -315,11 +327,16 @@ extension Archive {
         _ = try Data.write(chunk: endOfCentralDirRecord.data, to: self.archiveFile)
     }
 
-    private func replaceCurrentArchiveWithArchive(at URL: URL) throws {
+    func replaceCurrentArchiveWithArchive(at URL: URL) throws {
         fclose(self.archiveFile)
         let fileManager = FileManager()
         #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
-            _ = try fileManager.replaceItemAt(self.url, withItemAt: URL)
+            do {
+                _ = try fileManager.replaceItemAt(self.url, withItemAt: URL)
+            } catch {
+                _ = try fileManager.removeItem(at: self.url)
+                _ = try fileManager.moveItem(at: URL, to: self.url)
```

Comment:
We should try to keep this method private because it changes internal state. I assume you exposed it to make it testable. Maybe we can exercise this codepath by triggering the calling method from the test case?

Reply:
`remove/move` will succeed when `replace` fails in the case when the source and destination URLs represent content on different volumes.

Reply:
Yes, I did expose it so it could be tested... `private` feels is a bit aggressive, but if you maintain it should be `private` I'll have to look into trying to test this through a calling method.

Reply:
I looked at this a bit and it would be difficult to attempt to test this method through the caller. Essentially, this is called when attempting to remove an entry from a compressed archive. A temp archive is created in a temporary directory. To test the cross-link situation the temporary archive would have to reside on a different volume than the archive being manipulated, however, the new temporary directory method has been explicitly changed to create the temporary directory on the same volume. We would have to manufacture a failover mechanism such that the temporary directory creation fell back to the `NSTemporaryDirectory()` usage, while placing the main archive on a separate volume. That sounds like a significant effort that I'm not sure I'm able to commit to, considering the alternative is to leave this method exposed for testing.

Reply:
OK. No worries. I will have a look (hopefully tomorrow) if we can refactor that a bit or otherwise we can keep those methods `internal` for now.